code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,176 @@
+// ***************************************
+// *********** Drain blood
+// ***************************************
+/datum/action/xeno_action/activable/drain
+	name = "Drain"
+	action_icon_state = "drain"
+	mechanics_text = "Hold a marine and drain some of their blood if successful."
+	ability_name = "drain"
+	cooldown_timer = 15 SECONDS
+	plasma_cost = 0
+
+/datum/action/xeno_action/activable/drain/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!iscarbon(A))
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, span_xenowarning("This won't do!"))
+		return FALSE
```

Comment:
here doesnt check for silent and following too

also "this wont do" should be more descriptive

Reply:
changed